{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "3b831e4c-cb51-4bf9-8ad5-1c651c8e1eea",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: dfcx-scrapi in /home/trauco/bc-records/venv/lib/python3.10/site-packages (1.9.0)\n",
      "Requirement already satisfied: google-cloud-dialogflow-cx in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from dfcx-scrapi) (1.28.0)\n",
      "Requirement already satisfied: google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.0 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-cloud-dialogflow-cx->dfcx-scrapi) (2.14.0)\n",
      "Requirement already satisfied: proto-plus<2.0.0dev,>=1.22.0 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-cloud-dialogflow-cx->dfcx-scrapi) (1.22.3)\n",
      "Requirement already satisfied: protobuf!=3.20.0,!=3.20.1,!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<5.0.0dev,>=3.19.5 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-cloud-dialogflow-cx->dfcx-scrapi) (4.25.1)\n",
      "Requirement already satisfied: google-auth<3.0.dev0,>=2.14.1 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.0->google-cloud-dialogflow-cx->dfcx-scrapi) (2.23.4)\n",
      "Requirement already satisfied: requests<3.0.0.dev0,>=2.18.0 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.0->google-cloud-dialogflow-cx->dfcx-scrapi) (2.31.0)\n",
      "Requirement already satisfied: googleapis-common-protos<2.0.dev0,>=1.56.2 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.0->google-cloud-dialogflow-cx->dfcx-scrapi) (1.61.0)\n",
      "Requirement already satisfied: grpcio<2.0dev,>=1.33.2 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.0->google-cloud-dialogflow-cx->dfcx-scrapi) (1.59.3)\n",
      "Requirement already satisfied: grpcio-status<2.0.dev0,>=1.33.2 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.0->google-cloud-dialogflow-cx->dfcx-scrapi) (1.59.3)\n",
      "Requirement already satisfied: cachetools<6.0,>=2.0.0 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-auth<3.0.dev0,>=2.14.1->google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.0->google-cloud-dialogflow-cx->dfcx-scrapi) (5.3.1)\n",
      "Requirement already satisfied: pyasn1-modules>=0.2.1 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-auth<3.0.dev0,>=2.14.1->google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.0->google-cloud-dialogflow-cx->dfcx-scrapi) (0.3.0)\n",
      "Requirement already satisfied: rsa<5,>=3.1.4 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-auth<3.0.dev0,>=2.14.1->google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.0->google-cloud-dialogflow-cx->dfcx-scrapi) (4.9)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from requests<3.0.0.dev0,>=2.18.0->google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.0->google-cloud-dialogflow-cx->dfcx-scrapi) (3.3.0)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from requests<3.0.0.dev0,>=2.18.0->google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.0->google-cloud-dialogflow-cx->dfcx-scrapi) (2.0.6)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from requests<3.0.0.dev0,>=2.18.0->google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.0->google-cloud-dialogflow-cx->dfcx-scrapi) (2023.7.22)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from requests<3.0.0.dev0,>=2.18.0->google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.0->google-cloud-dialogflow-cx->dfcx-scrapi) (3.4)\n",
      "Requirement already satisfied: pyasn1<0.6.0,>=0.4.6 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from pyasn1-modules>=0.2.1->google-auth<3.0.dev0,>=2.14.1->google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.0->google-cloud-dialogflow-cx->dfcx-scrapi) (0.5.0)\n",
      "Note: you may need to restart the kernel to use updated packages.\n",
      "Requirement already satisfied: dfcx-scrapi in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from -r ./requirements.txt (line 1)) (1.9.0)\n",
      "Requirement already satisfied: google-cloud-dialogflow-cx in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from -r ./requirements.txt (line 2)) (1.28.0)\n",
      "Requirement already satisfied: google-auth in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from -r ./requirements.txt (line 3)) (2.23.4)\n",
      "Collecting google-oauth\n",
      "  Using cached google-oauth-1.0.1.tar.gz (5.6 kB)\n",
      "  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hRequirement already satisfied: oauth2client in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from -r ./requirements.txt (line 5)) (4.1.3)\n",
      "Collecting pyparsing==2.4.7\n",
      "  Using cached pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)\n",
      "Requirement already satisfied: pandas in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from -r ./requirements.txt (line 7)) (2.1.1)\n",
      "Requirement already satisfied: tabulate in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from -r ./requirements.txt (line 8)) (0.9.0)\n",
      "Collecting gspread==5.10.0\n",
      "  Using cached gspread-5.10.0-py3-none-any.whl (44 kB)\n",
      "Requirement already satisfied: gspread_dataframe in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from -r ./requirements.txt (line 10)) (3.3.1)\n",
      "Requirement already satisfied: numpy in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from -r ./requirements.txt (line 11)) (1.26.0)\n",
      "Requirement already satisfied: requests in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from -r ./requirements.txt (line 12)) (2.31.0)\n",
      "Collecting pylint==2.8.3\n",
      "  Using cached pylint-2.8.3-py3-none-any.whl (357 kB)\n",
      "Collecting pytest==6.0.2\n",
      "  Using cached pytest-6.0.2-py3-none-any.whl (270 kB)\n",
      "Collecting pytest-cov==2.11.1\n",
      "  Using cached pytest_cov-2.11.1-py2.py3-none-any.whl (20 kB)\n",
      "Collecting pytest-xdist==2.1.0\n",
      "  Using cached pytest_xdist-2.1.0-py3-none-any.whl (36 kB)\n",
      "Collecting pyyaml\n",
      "  Using cached PyYAML-6.0.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (705 kB)\n",
      "Collecting torch\n",
      "  Using cached torch-2.1.1-cp310-cp310-manylinux1_x86_64.whl (670.2 MB)\n",
      "Collecting transformers\n",
      "  Using cached transformers-4.35.2-py3-none-any.whl (7.9 MB)\n",
      "Collecting sentencepiece\n",
      "  Using cached sentencepiece-0.1.99-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.3 MB)\n",
      "Requirement already satisfied: google-auth-oauthlib>=0.4.1 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from gspread==5.10.0->-r ./requirements.txt (line 9)) (1.1.0)\n",
      "Collecting toml>=0.7.1\n",
      "  Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)\n",
      "Collecting mccabe<0.7,>=0.6\n",
      "  Using cached mccabe-0.6.1-py2.py3-none-any.whl (8.6 kB)\n",
      "Collecting astroid==2.5.6\n",
      "  Using cached astroid-2.5.6-py3-none-any.whl (219 kB)\n",
      "Collecting isort<6,>=4.2.5\n",
      "  Using cached isort-5.12.0-py3-none-any.whl (91 kB)\n",
      "Collecting py>=1.8.2\n",
      "  Using cached py-1.11.0-py2.py3-none-any.whl (98 kB)\n",
      "Collecting pluggy<1.0,>=0.12\n",
      "  Using cached pluggy-0.13.1-py2.py3-none-any.whl (18 kB)\n",
      "Collecting attrs>=17.4.0\n",
      "  Using cached attrs-23.1.0-py3-none-any.whl (61 kB)\n",
      "Requirement already satisfied: packaging in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from pytest==6.0.2->-r ./requirements.txt (line 14)) (23.2)\n",
      "Collecting more-itertools>=4.0.0\n",
      "  Using cached more_itertools-10.1.0-py3-none-any.whl (55 kB)\n",
      "Collecting iniconfig\n",
      "  Using cached iniconfig-2.0.0-py3-none-any.whl (5.9 kB)\n",
      "Collecting coverage>=5.2.1\n",
      "  Using cached coverage-7.3.2-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (227 kB)\n",
      "Collecting pytest-forked\n",
      "  Using cached pytest_forked-1.6.0-py3-none-any.whl (4.9 kB)\n",
      "Collecting execnet>=1.1\n",
      "  Using cached execnet-2.0.2-py3-none-any.whl (37 kB)\n",
      "Collecting wrapt<1.13,>=1.11\n",
      "  Using cached wrapt-1.12.1.tar.gz (27 kB)\n",
      "  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hCollecting lazy-object-proxy>=1.4.0\n",
      "  Using cached lazy_object_proxy-1.9.0-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (63 kB)\n",
      "Requirement already satisfied: protobuf!=3.20.0,!=3.20.1,!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<5.0.0dev,>=3.19.5 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-cloud-dialogflow-cx->-r ./requirements.txt (line 2)) (4.25.1)\n",
      "Requirement already satisfied: google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.0 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-cloud-dialogflow-cx->-r ./requirements.txt (line 2)) (2.14.0)\n",
      "Requirement already satisfied: proto-plus<2.0.0dev,>=1.22.0 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-cloud-dialogflow-cx->-r ./requirements.txt (line 2)) (1.22.3)\n",
      "Requirement already satisfied: pyasn1-modules>=0.2.1 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-auth->-r ./requirements.txt (line 3)) (0.3.0)\n",
      "Requirement already satisfied: cachetools<6.0,>=2.0.0 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-auth->-r ./requirements.txt (line 3)) (5.3.1)\n",
      "Requirement already satisfied: rsa<5,>=3.1.4 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-auth->-r ./requirements.txt (line 3)) (4.9)\n",
      "Collecting pyopenssl>=0.11\n",
      "  Using cached pyOpenSSL-23.3.0-py3-none-any.whl (58 kB)\n",
      "Requirement already satisfied: six in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-oauth->-r ./requirements.txt (line 4)) (1.16.0)\n",
      "Requirement already satisfied: httplib2>=0.9.1 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from oauth2client->-r ./requirements.txt (line 5)) (0.22.0)\n",
      "Requirement already satisfied: pyasn1>=0.1.7 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from oauth2client->-r ./requirements.txt (line 5)) (0.5.0)\n",
      "Requirement already satisfied: pytz>=2020.1 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from pandas->-r ./requirements.txt (line 7)) (2023.3.post1)\n",
      "Requirement already satisfied: tzdata>=2022.1 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from pandas->-r ./requirements.txt (line 7)) (2023.3)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from pandas->-r ./requirements.txt (line 7)) (2.8.2)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from requests->-r ./requirements.txt (line 12)) (2023.7.22)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from requests->-r ./requirements.txt (line 12)) (2.0.6)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from requests->-r ./requirements.txt (line 12)) (3.3.0)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from requests->-r ./requirements.txt (line 12)) (3.4)\n",
      "Collecting fsspec\n",
      "  Downloading fsspec-2023.10.0-py3-none-any.whl (166 kB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m166.4/166.4 KB\u001b[0m \u001b[31m612.3 kB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m kB/s\u001b[0m eta \u001b[36m0:00:01\u001b[0m:01\u001b[0m\n",
      "\u001b[?25hCollecting nvidia-nvtx-cu12==12.1.105\n",
      "  Downloading nvidia_nvtx_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (99 kB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m99.1/99.1 KB\u001b[0m \u001b[31m1.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m[31m8.0 MB/s\u001b[0m eta \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hCollecting nvidia-cuda-runtime-cu12==12.1.105\n",
      "  Downloading nvidia_cuda_runtime_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (823 kB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m823.6/823.6 KB\u001b[0m \u001b[31m490.1 kB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m1m485.7 kB/s\u001b[0m eta \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hCollecting networkx\n",
      "  Downloading networkx-3.2.1-py3-none-any.whl (1.6 MB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.6/1.6 MB\u001b[0m \u001b[31m1.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m MB/s\u001b[0m eta \u001b[36m0:00:01\u001b[0m:01\u001b[0m0m\n",
      "\u001b[?25hCollecting jinja2\n",
      "  Using cached Jinja2-3.1.2-py3-none-any.whl (133 kB)\n",
      "Collecting nvidia-cuda-nvrtc-cu12==12.1.105\n",
      "  Downloading nvidia_cuda_nvrtc_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (23.7 MB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m23.7/23.7 MB\u001b[0m \u001b[31m2.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0mm eta \u001b[36m0:00:01\u001b[0m[36m0:00:01\u001b[0m\n",
      "\u001b[?25hCollecting nvidia-cublas-cu12==12.1.3.1\n",
      "  Downloading nvidia_cublas_cu12-12.1.3.1-py3-none-manylinux1_x86_64.whl (410.6 MB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m410.6/410.6 MB\u001b[0m \u001b[31m3.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0mm eta \u001b[36m0:00:01\u001b[0m[36m0:00:03\u001b[0m\n",
      "\u001b[?25hCollecting nvidia-cusparse-cu12==12.1.0.106\n",
      "  Using cached nvidia_cusparse_cu12-12.1.0.106-py3-none-manylinux1_x86_64.whl (196.0 MB)\n",
      "Collecting nvidia-cusolver-cu12==11.4.5.107\n",
      "  Downloading nvidia_cusolver_cu12-11.4.5.107-py3-none-manylinux1_x86_64.whl (124.2 MB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m124.2/124.2 MB\u001b[0m \u001b[31m4.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0mm eta \u001b[36m0:00:01\u001b[0m[36m0:00:01\u001b[0m\n",
      "\u001b[?25hCollecting triton==2.1.0\n",
      "  Downloading triton-2.1.0-0-cp310-cp310-manylinux2014_x86_64.manylinux_2_17_x86_64.whl (89.2 MB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m89.2/89.2 MB\u001b[0m \u001b[31m4.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0mm eta \u001b[36m0:00:01\u001b[0m[36m0:00:01\u001b[0m\n",
      "\u001b[?25hCollecting nvidia-cufft-cu12==11.0.2.54\n",
      "  Using cached nvidia_cufft_cu12-11.0.2.54-py3-none-manylinux1_x86_64.whl (121.6 MB)\n",
      "Collecting filelock\n",
      "  Downloading filelock-3.13.1-py3-none-any.whl (11 kB)\n",
      "Collecting nvidia-cudnn-cu12==8.9.2.26\n",
      "  Downloading nvidia_cudnn_cu12-8.9.2.26-py3-none-manylinux1_x86_64.whl (731.7 MB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m731.7/731.7 MB\u001b[0m \u001b[31m2.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0mm eta \u001b[36m0:00:01\u001b[0m[36m0:00:05\u001b[0m\n",
      "\u001b[?25hCollecting nvidia-curand-cu12==10.3.2.106\n",
      "  Using cached nvidia_curand_cu12-10.3.2.106-py3-none-manylinux1_x86_64.whl (56.5 MB)\n",
      "Collecting typing-extensions\n",
      "  Using cached typing_extensions-4.8.0-py3-none-any.whl (31 kB)\n",
      "Collecting sympy\n",
      "  Using cached sympy-1.12-py3-none-any.whl (5.7 MB)\n",
      "Collecting nvidia-nccl-cu12==2.18.1\n",
      "  Using cached nvidia_nccl_cu12-2.18.1-py3-none-manylinux1_x86_64.whl (209.8 MB)\n",
      "Collecting nvidia-cuda-cupti-cu12==12.1.105\n",
      "  Using cached nvidia_cuda_cupti_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (14.1 MB)\n",
      "Collecting nvidia-nvjitlink-cu12\n",
      "  Downloading nvidia_nvjitlink_cu12-12.3.101-py3-none-manylinux1_x86_64.whl (20.5 MB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m20.5/20.5 MB\u001b[0m \u001b[31m2.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0mm eta \u001b[36m0:00:01\u001b[0m[36m0:00:01\u001b[0m\n",
      "\u001b[?25hCollecting huggingface-hub<1.0,>=0.16.4\n",
      "  Downloading huggingface_hub-0.19.4-py3-none-any.whl (311 kB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m311.7/311.7 KB\u001b[0m \u001b[31m2.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m[31m2.7 MB/s\u001b[0m eta \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hCollecting tokenizers<0.19,>=0.14\n",
      "  Downloading tokenizers-0.15.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.8 MB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.8/3.8 MB\u001b[0m \u001b[31m2.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0mm eta \u001b[36m0:00:01\u001b[0m[36m0:00:01\u001b[0m\n",
      "\u001b[?25hCollecting regex!=2019.12.17\n",
      "  Downloading regex-2023.10.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (773 kB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m773.9/773.9 KB\u001b[0m \u001b[31m3.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0mm eta \u001b[36m0:00:01\u001b[0m0:01\u001b[0m:01\u001b[0m\n",
      "\u001b[?25hCollecting tqdm>=4.27\n",
      "  Downloading tqdm-4.66.1-py3-none-any.whl (78 kB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m78.3/78.3 KB\u001b[0m \u001b[31m6.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hCollecting safetensors>=0.3.1\n",
      "  Downloading safetensors-0.4.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.3 MB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.3/1.3 MB\u001b[0m \u001b[31m5.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0mm eta \u001b[36m0:00:01\u001b[0m[36m0:00:01\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: googleapis-common-protos<2.0.dev0,>=1.56.2 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.0->google-cloud-dialogflow-cx->-r ./requirements.txt (line 2)) (1.61.0)\n",
      "Requirement already satisfied: grpcio<2.0dev,>=1.33.2 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.0->google-cloud-dialogflow-cx->-r ./requirements.txt (line 2)) (1.59.3)\n",
      "Requirement already satisfied: grpcio-status<2.0.dev0,>=1.33.2 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.10.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,!=2.9.*,<3.0.0dev,>=1.34.0->google-cloud-dialogflow-cx->-r ./requirements.txt (line 2)) (1.59.3)\n",
      "Requirement already satisfied: requests-oauthlib>=0.7.0 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from google-auth-oauthlib>=0.4.1->gspread==5.10.0->-r ./requirements.txt (line 9)) (1.3.1)\n",
      "Collecting cryptography<42,>=41.0.5\n",
      "  Downloading cryptography-41.0.5-cp37-abi3-manylinux_2_28_x86_64.whl (4.4 MB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m4.4/4.4 MB\u001b[0m \u001b[31m6.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m[36m0:00:01\u001b[0m[36m0:00:01\u001b[0m:01\u001b[0mm\n",
      "\u001b[?25hCollecting MarkupSafe>=2.0\n",
      "  Using cached MarkupSafe-2.1.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (25 kB)\n",
      "Collecting mpmath>=0.19\n",
      "  Downloading mpmath-1.3.0-py3-none-any.whl (536 kB)\n",
      "\u001b[2K     \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m536.2/536.2 KB\u001b[0m \u001b[31m2.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m[31m2.7 MB/s\u001b[0m eta \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hCollecting cffi>=1.12\n",
      "  Using cached cffi-1.16.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (443 kB)\n",
      "Requirement already satisfied: oauthlib>=3.0.0 in /home/trauco/bc-records/venv/lib/python3.10/site-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib>=0.4.1->gspread==5.10.0->-r ./requirements.txt (line 9)) (3.2.2)\n",
      "Collecting pycparser\n",
      "  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)\n",
      "Using legacy 'setup.py install' for google-oauth, since package 'wheel' is not installed.\n",
      "Using legacy 'setup.py install' for wrapt, since package 'wheel' is not installed.\n",
      "Installing collected packages: wrapt, sentencepiece, mpmath, mccabe, typing-extensions, tqdm, toml, sympy, safetensors, regex, pyyaml, pyparsing, pycparser, py, pluggy, nvidia-nvtx-cu12, nvidia-nvjitlink-cu12, nvidia-nccl-cu12, nvidia-curand-cu12, nvidia-cufft-cu12, nvidia-cuda-runtime-cu12, nvidia-cuda-nvrtc-cu12, nvidia-cuda-cupti-cu12, nvidia-cublas-cu12, networkx, more-itertools, MarkupSafe, lazy-object-proxy, isort, iniconfig, fsspec, filelock, execnet, coverage, attrs, triton, pytest, nvidia-cusparse-cu12, nvidia-cudnn-cu12, jinja2, huggingface-hub, cffi, astroid, tokenizers, pytest-forked, pytest-cov, pylint, nvidia-cusolver-cu12, cryptography, transformers, torch, pytest-xdist, pyopenssl, gspread, google-oauth\n",
      "  Running setup.py install for wrapt ... \u001b[?25ldone\n",
      "\u001b[?25h  Attempting uninstall: pyparsing\n",
      "    Found existing installation: pyparsing 3.1.1\n",
      "    Uninstalling pyparsing-3.1.1:\n",
      "      Successfully uninstalled pyparsing-3.1.1\n",
      "  Attempting uninstall: gspread\n",
      "    Found existing installation: gspread 5.11.3\n",
      "    Uninstalling gspread-5.11.3:\n",
      "      Successfully uninstalled gspread-5.11.3\n",
      "  Running setup.py install for google-oauth ... \u001b[?25ldone\n",
      "\u001b[?25hSuccessfully installed MarkupSafe-2.1.3 astroid-2.5.6 attrs-23.1.0 cffi-1.16.0 coverage-7.3.2 cryptography-41.0.5 execnet-2.0.2 filelock-3.13.1 fsspec-2023.10.0 google-oauth-1.0.1 gspread-5.10.0 huggingface-hub-0.19.4 iniconfig-2.0.0 isort-5.12.0 jinja2-3.1.2 lazy-object-proxy-1.9.0 mccabe-0.6.1 more-itertools-10.1.0 mpmath-1.3.0 networkx-3.2.1 nvidia-cublas-cu12-12.1.3.1 nvidia-cuda-cupti-cu12-12.1.105 nvidia-cuda-nvrtc-cu12-12.1.105 nvidia-cuda-runtime-cu12-12.1.105 nvidia-cudnn-cu12-8.9.2.26 nvidia-cufft-cu12-11.0.2.54 nvidia-curand-cu12-10.3.2.106 nvidia-cusolver-cu12-11.4.5.107 nvidia-cusparse-cu12-12.1.0.106 nvidia-nccl-cu12-2.18.1 nvidia-nvjitlink-cu12-12.3.101 nvidia-nvtx-cu12-12.1.105 pluggy-0.13.1 py-1.11.0 pycparser-2.21 pylint-2.8.3 pyopenssl-23.3.0 pyparsing-2.4.7 pytest-6.0.2 pytest-cov-2.11.1 pytest-forked-1.6.0 pytest-xdist-2.1.0 pyyaml-6.0.1 regex-2023.10.3 safetensors-0.4.0 sentencepiece-0.1.99 sympy-1.12 tokenizers-0.15.0 toml-0.10.2 torch-2.1.1 tqdm-4.66.1 transformers-4.35.2 triton-2.1.0 typing-extensions-4.8.0 wrapt-1.12.1\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "%pip install dfcx-scrapi\n",
    "%pip install -r ./requirements.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "81bc06dc-98a0-4c3a-9a94-d878f7a1379d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from dfcx_scrapi.core.agents import Agents\n",
    "from dfcx_scrapi.core.intents import Intents\n",
    "from dfcx_scrapi.core.flows import Flows\n",
    "from dfcx_scrapi.core.pages import Pages\n",
    "from dfcx_scrapi.tools.dataframe_functions import DataframeFunctions\n",
    "from dfcx_scrapi.tools.maker_util import MakerUtil\n",
    "\n",
    "#Input here your Service Account Key path\n",
    "creds_path = '/home/trauco/.config/secrets/cx_admin_sa.json'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ffed009c-9f20-4d5d-834d-b31d9eeffd04",
   "metadata": {},
   "outputs": [],
   "source": [
    "# First we will instantiate our Agent object\n",
    "a = Agents(creds_path=creds_path)\n",
    "\n",
    "# Next, we will set some variables for our agent creation or retrieval args\n",
    "project_id = 'bot1-404120'\n",
    "display_name = 'bc-records'\n",
    "gcp_region = 'global' #(eg. 'global', 'us-cenral1', etc... See region availability for DFCX at https://cloud.google.com/dialogflow/cx/docs/concept/region\n",
    "\n",
    "# Then we will call the `create_agent` or get_agent method (if agent already exists) and capture the result in a var call `my_agent`\n",
    "#Option 1: If agent does not exist\n",
    "my_agent = a.create_agent(project_id, display_name, gcp_region)\n",
    "\n",
    "#Option 2: If agent already exists\n",
    "#The agent ID must be entered as a string \"projects/<PROJECT_ID>/locations/<GCP_REGION>/agents/<AGENT_ID>\"\n",
    "# my_agent = a.get_agent(\"projects/<PROJECT_ID>/locations/<GCP_REGION>/agents/<AGENT_ID>\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "dc7bf734-bbc7-4fc3-8a44-b55990cec79a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Progress(1/1)[------------------------------------------------->] 100.0%\r"
     ]
    }
   ],
   "source": [
    "# First, we will instantiate our DataframeFunctions (dffx) object\n",
    "dffx = DataframeFunctions(creds_path=creds_path)\n",
    "\n",
    "# Next, we will read in our sample CSV with Intent training phrases data as well as the entities used into 2 distinct Pandas DataFrames\n",
    "df = pd.read_csv('intent_sample_with_parts.csv')\n",
    "params_df = pd.read_csv('intent_sample_params.csv')\n",
    "\n",
    "# Finally, we will use `dffx` to push our Intents to our Agent\n",
    "#If intents do not exist, use bulk_create_intent_from_dataframe\n",
    "my_intents = dffx.bulk_create_intent_from_dataframe(my_agent.name, df, params_df, update_flag=True, mode=\"advanced\")\n",
    "\n",
    "#If intent already exist, use bulk_update_intents_from_dataframe\n",
    "#my_intents = dffx.bulk_update_intents_from_dataframe(my_agent.name, df, params_df, update_flag=True, mode=\"advanced\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cda681f1-516e-416f-9468-30521489b160",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
